src/main/java/com/epam/test_generator/services/StepService.java

```diff
@@ -113,4 +113,30 @@ public void removeStep(Long suitId, Long caseId, Long stepId) {
         caze.getSteps().remove(step);
         stepDAO.delete(stepId);
     }
+
+    /**
+     * This service method is specialized for 'adding' 'editing' and 'deleting' a list of steps
+     *
+     * For selecting required action uses "action" field from DTO object
+     *
+     * @param steps array list of steps
+     */
+    public void cascadeUpdateSteps(long suitId, long caseId, List<StepDTO> steps) {
+        for (StepDTO stepDTO: steps) {
+            String action = stepDTO.getAction();
+            if(action != null){
+                switch(action.toLowerCase()) {
+                    case "update":
+                        updateStep(suitId, caseId, stepDTO.getId(), stepDTO);
+                        break;
+                    case "create":
+                        addStepToCase(suitId, caseId, stepDTO);
+                        break;
+                    case "delete":
+                        removeStep(suitId, caseId, stepDTO.getId());
+                        break;
```

Comment:
default is required here in case value is unknown.

Reply:
Done